pulser-core/pulser/devices/_device_datacls.py

```diff
@@ -538,6 +538,47 @@ def calibrated_register_layouts(self) -> dict[str, RegisterLayout]:
         """Register layouts already calibrated on this device."""
         return {str(layout): layout for layout in self.pre_calibrated_layouts}
 
+    def is_calibrated_layout(self, register_layout: RegisterLayout) -> bool:
+        """Checks whether a layout is within the calibrated layouts.
+
+        Args:
+            register_layout: the RegisterLayout to check.
+
+        Returns:
+            True if register_layout is found among calibrated_register_layouts,
+            False otherwise.
+        """
+        for (
+            name,
+            calibrated_layout,
+        ) in self.calibrated_register_layouts.items():
+            if register_layout == calibrated_layout:
+                return True
+        return False
+
+    def register_is_from_calibrated_layout(
+        self, register: BaseRegister | MappableRegister
+    ) -> bool:
+        """Checks whether a register was constructed from a calibrated layout.
+
+        If the register is a BaseRegister, checks that it has a layout. If so,
+        or if it is a MappableRegister, check that its layout is within the
+        calibrated layouts.
+
+        Args:
+            register_layout: the Register or MappableRegister to check.
+
+        Returns:
+            True if register has a layout and it is found among
+            calibrated_register_layouts, False otherwise.
+        """
+        if (
+            isinstance(register, BaseRegister)
+            and register._layout_info is None
```

Comment:
You can check directly
```suggestion
            and register.layout is None
```